src/ring_theory/jacobson.lean

```diff
@@ -378,128 +395,164 @@ begin
-  let J : ideal (polynomial I'.quotient) := I.map f,
-  have hIJ : I = J.comap f := begin
-    rw comap_map_of_surjective f hf,
+  let P' : ideal R := P.comap C,
+  haveI : P'.is_prime := comap_is_prime C P,
+  let f : polynomial R →+* polynomial P'.quotient := polynomial.map_ring_hom (quotient.mk P'),
+  have hf : function.surjective f := map_surjective (quotient.mk P') quotient.mk_surjective,
+  have hPJ : P = (P.map f).comap f,
+  { rw comap_map_of_surjective _ hf,
     refine le_antisymm (le_sup_left_of_le le_rfl) (sup_le le_rfl _),
-    refine λ p hp, polynomial_mem_ideal_of_coeff_mem_ideal I p (λ n, _),
-    rw ← quotient.eq_zero_iff_mem,
-    rw [mem_comap, ideal.mem_bot, polynomial.ext_iff] at hp,
-    simpa using hp n,
-  end,
-  haveI : J.is_maximal := or.rec_on (map_eq_top_or_is_maximal_of_surjective f hf hI)
-    (λ h, absurd (trans (h ▸ hIJ : I = comap f ⊤) comap_top : I = ⊤) hI.1) id,
-  let i' : I.quotient →+* J.quotient := quotient_map J f le_comap_map,
-  have hi' : function.injective i' := quotient_map_injective' (le_of_eq hIJ.symm),
-  let ϕ : R →+* I.quotient := (quotient.mk I).comp C,
-  let ϕ' : I'.quotient →+* J.quotient := (quotient.mk J).comp C,
-  refine ring_hom.is_integral_tower_bot_of_is_integral ϕ i' hi' _,
-  refine (ring_hom.ext (λ _, by simp) : ϕ'.comp i = i'.comp ϕ) ▸ ring_hom.is_integral_trans i ϕ'
-    (i.is_integral_of_surjective quotient.mk_surjective)
-    (quotient_mk_comp_C_is_integral_of_jacobson' J (λ x hx, _)),
-  obtain ⟨z, rfl⟩ := quotient.mk_surjective x,
-  rwa [quotient.eq_zero_iff_mem, mem_comap, hIJ, mem_comap, coe_map_ring_hom, map_C],
+    refine λ p hp, polynomial_mem_ideal_of_coeff_mem_ideal P p (λ n, quotient.eq_zero_iff_mem.mp _),
+    simpa only [coeff_map, coe_map_ring_hom] using (polynomial.ext_iff.mp hp) n },
+  refine ring_hom.is_integral_tower_bot_of_is_integral _ _ (injective_quotient_le_comap_map P) _,
+  rw ← quotient_mk_maps_eq,
+  refine ring_hom.is_integral_trans _ _
+    ((quotient.mk P').is_integral_of_surjective quotient.mk_surjective) _,
+  apply quotient_mk_comp_C_is_integral_of_jacobson' _ _ (λ x hx, _),
+  any_goals { exact ideal.is_jacobson_quotient },
+  { exact or.rec_on (map_eq_top_or_is_maximal_of_surjective f hf hP)
+    (λ h, absurd (trans (h ▸ hPJ : P = comap f ⊤) comap_top : P = ⊤) hP.1) id },
+  { obtain ⟨z, rfl⟩ := quotient.mk_surjective x,
+    rwa [quotient.eq_zero_iff_mem, mem_comap, hPJ, mem_comap, coe_map_ring_hom, map_C] }
 end
 
-lemma comp_C_integral_of_surjective_of_jacobson {R : Type*} [integral_domain R] [is_jacobson R]
+lemma is_maximal_comap_C_of_is_jacobson : (P.comap (C : R →+* polynomial R)).is_maximal :=
+begin
+  rw [← @mk_ker _ _ P, ring_hom.ker_eq_comap_bot, comap_comap],
+  exact is_maximal_comap_of_is_integral_of_is_maximal' _
+    (quotient_mk_comp_C_is_integral_of_jacobson P) ⊥ ((bot_quotient_is_maximal_iff _).mpr hP),
+end
+
+omit P hP
+
+lemma comp_C_integral_of_surjective_of_jacobson
```

Comment:
Any reason these lemmas changed order? Could you swap them back around to keep the diff small?